src/loki/route_action.cc

```diff
@@ -88,7 +88,7 @@ void loki_worker_t::route(Api& request) {
   // correlate the various locations to the underlying graph
   std::unordered_map<size_t, size_t> color_counts;
   try {
-    auto locations = PathLocation::fromPBF(options.locations());
+    auto locations = PathLocation::fromPBF(options.locations(), true);
```

Comment:
this little trick allows us to only care about outbound reach for origins and inbound reach for destinations. it lets us not cull otherwise good edges for these special case location candidates